src/components/CartIcon/CartIcon.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { AiOutlineShoppingCart } from 'react-icons/ai';
+import { Link } from 'react-router-dom';
+import styles from './CartIcon.module.css';
+import { useGlobalContext } from '../../context/Provider';
+
+function CartIcon() {
+    const { itemsInCart } = useGlobalContext();
+
+    const totalItems = itemsInCart
+        ? Object.values(itemsInCart).reduce(
+              (total, singleValue) => total + singleValue.quantity,
+              0
+          )
+        : 0;
```

Comment:
`itemsInCart` can be initialized to prevent errors and the assignment of `totalItems` can be simplified by removing the ternary, `reduce` will return the initial value if the array has no items:

```suggestion
    const { itemsInCart = {} } = useGlobalContext();

    const totalItems = Object.values(itemsInCart).reduce(
      (total, { quantity }) => total + quantity, 0
    );
```